parquet-cli/src/test/java/org/apache/parquet/cli/commands/ColumnSizeCommandTest.java

```diff
@@ -0,0 +1,99 @@
+ * under the License.
+ */
+package org.apache.parquet.cli.commands;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.example.data.simple.SimpleGroup;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.example.ExampleParquetWriter;
+import org.apache.parquet.hadoop.example.GroupWriteSupport;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.PrimitiveType;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Random;
+
+import static org.apache.parquet.schema.PrimitiveType.PrimitiveTypeName.INT32;
+import static org.apache.parquet.schema.PrimitiveType.PrimitiveTypeName.INT64;
+import static org.apache.parquet.schema.Type.Repetition.REQUIRED;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ColumnSizeCommandTest extends ParquetFileTest {
+
+  private final int numRecord = 10000;
+  private ColumnSizeCommand command = new ColumnSizeCommand(createLogger());
+  private Configuration conf = new Configuration();
+
+  @Test
+  public void testColumnSizeCommand() throws IOException {
+    File file = parquetFile();
+    ColumnSizeCommand command = new ColumnSizeCommand(createLogger());
+    command.target = file.getAbsolutePath();
+    command.setConf(new Configuration());
+    Assert.assertEquals(0, command.run());
+  }
+
+  @Test
+  public void testColumnSize() throws Exception {
+    String inputFile = createParquetFile("input");
+    Map<String, Long> columnSizeInBytes = command.getColumnSizeInBytes(new Path(inputFile));
+    assertEquals(columnSizeInBytes.size(), 2);
+    assertTrue(columnSizeInBytes.get("DocId") > columnSizeInBytes.get("Num"));
+    Map<String, Float> columnRatio = command.getColumnRatio(columnSizeInBytes);
+    assertTrue(columnRatio.get("DocId") > columnRatio.get("Num"));
+  }
+
+  private String createParquetFile(String prefix) throws IOException {
+    MessageType schema = new MessageType("schema",
+      new PrimitiveType(REQUIRED, INT64, "DocId"),
+      new PrimitiveType(REQUIRED, INT32, "Num"));
+
+    conf.set(GroupWriteSupport.PARQUET_EXAMPLE_SCHEMA, schema.toString());
+
+    String file = createTempFile(prefix);
+    ExampleParquetWriter.Builder builder = ExampleParquetWriter.builder(new Path(file)).withConf(conf);
+    Random rnd = new Random();
+    try (ParquetWriter writer = builder.build()) {
+      for (int i = 0; i < numRecord; i++) {
+        SimpleGroup g = new SimpleGroup(schema);
+        g.add("DocId", rnd.nextLong());
+        g.add("Num", rnd.nextInt());
+        writer.write(g);
+      }
+    }
+
+    return file;
+  }
+
+  private static String createTempFile(String prefix) {
+    try {
+      return Files.createTempDirectory(prefix).toAbsolutePath().toString() + "/test.parquet";
```

Comment:
See comment in TestColumnSizeCommand.java 👍

Reply:
Sounds good!